spec/models/miq_group_spec.rb

```diff
@@ -1,6 +1,14 @@
 RSpec.describe MiqGroup do
   include Spec::Support::ArelHelper
 
+  let(:miq_widget) { FactoryBot.create(:miq_widget) }
+
+  let(:set_data) do
```

Comment:
Not a big fan of directly manipulating the set hash - here or in the controllers either. Maybe using add_member or the transient traits would be better